input_pipeline/data_extraction.py

```diff
@@ -0,0 +1,82 @@
+from os import makedirs, path
+from typing import Tuple, List, Dict
+from oracle import generate_sequence_of_actions
+import pickle as js
+
+class DataSet(object):
+
+  def __init__(self, lines: List[str]):
+    self.sentences: List[Sentence] = []
+    self.tokens_vocab: Dict[str, int] = {}
+    self.relations_vocab: Dict[str, int] = {}
+
+    entry_lines = []
+    for line in lines:
+      line = line.strip()
+      if line:
+        entry_lines.append(line)
+      elif entry_lines:
+        self.sentences.append(Sentence(entry_lines, self))
+        entry_lines.clear()
+    if entry_lines:
+      self.sentences.append(Sentence(entry_lines, self))
+
+class Sentence(object):
```

Comment:
What about DataSetEntry? It's not a sentence per say, it also contains dependency tree information